### PR TITLE
PARQUET-1280: [parquet-protobuf] Use maven protoc plugin

### PR DESCRIPTION
**Pull request by nandorKollar**
Use Maven plugin to generate test sources for Protobuf. Unfortunately I couldn't find any "official" plugin for this, not sure if the one I picked is the best choice.

### CODE REVIEW

File: parquet-protobuf/pom.xml [hunk 1]
```diff
@@ -148,28 +148,24 @@
...
+              <protocVersion>${protoc.version}</protocVersion>
+              <addProtoSources>all</addProtoSources>
+              <outputDirectory>${project.build.directory}/generated-test-sources/java</outputDirectory>
+              <inputDirectories>
+                <include>src/test/resources</include>
+              </inputDirectories>
+            </configuration>
           </execution>
         </executions>
       </plugin>
```
Comment (qinghui-xu) [inline parquet-protobuf/pom.xml]: Looks good to me

**Comment by qinghui-xu** [comment]
It might be interesting to merge it soon.

**Comment by qinghui-xu** [comment]
Let's merge it? @nandorKollar

**Comment by nandorKollar** [comment]
@qinghui-xu sorry, though I wish I could, but I can't merge this PR, since I'm not a committer on Parquet. In addition, AFAIK an other committers approval is needed to get this merged.

File: parquet-protobuf/pom.xml [hunk 2]
```diff
@@ -32,8 +32,8 @@
   <properties>
     <elephant-bird.version>4.4</elephant-bird.version>
     <protobuf.version>3.5.1</protobuf.version>
+    <protoc.version>3.5.1</protoc.version>
```
Comment (gszadovszky) [inline parquet-protobuf/pom.xml]: Why this new property is required next to the existing `protobuf.version` with the same value?

Comment (nandorKollar) [reply parquet-protobuf/pom.xml]: Good question. Can't think of any scenario where having two version (one for protobuf dependency and one for protoc plugin) would be beneficial.

Comment (gszadovszky) [reply parquet-protobuf/pom.xml]: So, it seems some cleanup might be required.
I think, it is good to use the same version number for the `protobuf-java` version and the protoc binary version. Let's use the one we already have.
We also have the `protoc.path` property which is not required anymore.

**Comment by gszadovszky** [comment]
@nandorKollar, this way we do not need the protobuf in our travis environment. Could you please remove the related parts

**Comment by qinghui-xu** [comment]
Thanks for merging this PR ;-)

**Pull request merged**